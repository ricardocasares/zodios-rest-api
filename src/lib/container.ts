export class Container<C> {
  constructor(private container: C) {}

  get(key: keyof C) {
    return this.container[key];
  }

  set(key: keyof C, value: C[keyof C]) {
    this.container[key] = value;
    return this;
  }
}
