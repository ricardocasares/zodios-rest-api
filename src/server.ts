import { serve, setup } from "swagger-ui-express";
import { temperature } from "./routes/temperature";
import { document } from "./openapi";
import { customCss } from "./swagger";

export const server = temperature.use("/docs", serve).use(
  "/docs",
  setup(document, {
    customCss,
    customSiteTitle: "Temperature API",
  })
);
