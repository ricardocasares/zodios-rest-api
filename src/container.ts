import { Container } from "./lib/container";
import { LocalTemperatureService } from "./services/localTemperature";
import type { TemperatureService } from "./models";

export const container = new Container({
  temp: new LocalTemperatureService() as TemperatureService,
});
