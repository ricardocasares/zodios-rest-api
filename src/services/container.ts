import { Container } from "../lib/container";
import { LocalTemperatureService } from "./localTemperature";
import type { TemperatureService } from "../lib/models";

export const container = new Container({
  temp: new LocalTemperatureService() as TemperatureService,
});
