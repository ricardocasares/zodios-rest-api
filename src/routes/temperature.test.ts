import supertest from "supertest";
import { server } from "../server";
import { container } from "../container";

const convert = jest
  .fn()
  .mockResolvedValueOnce(98.6)
  .mockResolvedValueOnce(37)
  .mockResolvedValue(37);

container.set("temp", { convert });

test("no parameters", () =>
  supertest(server)
    .get("/temperature")
    .expect(400)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("requires parameter [value]", () =>
  supertest(server)
    .get("/temperature")
    .query({ to: "F" })
    .expect(400)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("default parameter [to=F]", () =>
  supertest(server)
    .get("/temperature")
    .query({ value: 37 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("converts from C to F", () =>
  supertest(server)
    .get("/temperature")
    .query({ to: "C", value: 98.6 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("converts from F to C", () =>
  supertest(server)
    .get("/temperature")
    .query({ to: "C", value: 98.6 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));
