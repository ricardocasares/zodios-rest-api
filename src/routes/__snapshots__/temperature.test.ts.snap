// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts from C to F 1`] = `
{
  "scale": "C",
  "value": 37,
}
`;

exports[`converts from F to C 1`] = `
{
  "scale": "C",
  "value": 37,
}
`;

exports[`default parameter [to=F] 1`] = `
{
  "scale": "F",
  "value": 98.6,
}
`;

exports[`errors 1`] = `
{
  "message": "Internal error",
}
`;

exports[`no parameters 1`] = `
{
  "context": "query.value",
  "error": [
    {
      "code": "invalid_type",
      "expected": "number",
      "message": "Required",
      "path": [],
      "received": "undefined",
    },
  ],
}
`;

exports[`requires parameter [value] 1`] = `
{
  "context": "query.value",
  "error": [
    {
      "code": "invalid_type",
      "expected": "number",
      "message": "Required",
      "path": [],
      "received": "undefined",
    },
  ],
}
`;
